src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -58,11 +59,32 @@ private JVMStabilityInspector() {}
      */
     public static void inspectThrowable(Throwable t) throws OutOfMemoryError
     {
-        inspectThrowable(t, true);
+        inspectThrowable(t, null, true, true);
     }
 
+
     public static void inspectThrowable(Throwable t, boolean propagateOutOfMemory) throws OutOfMemoryError
     {
+        inspectThrowable(t, null, propagateOutOfMemory, true);
+    }
+
+    public static void inspectThrowable(Throwable t, Thread thread, boolean propagateOutOfMemory)
+    {
+        inspectThrowable(t, thread, propagateOutOfMemory, true);
+    }
+
+    private static void inspectThrowable(Throwable t, Thread thread, boolean propagateOutOfMemory, boolean shouldLog) throws OutOfMemoryError
+    {
+        if (shouldLog)
+        {
+            if (thread != null)
+                logger.error(String.format("Uncaught exception in thread %s", thread), t);
```

Comment:
with logger you can use `logger.error("Uncaught exception in thread {}", thread, t);`

Reply:
thanks, fixed.